isobus/include/isobus/isobus/isobus_shortcut_button_interface.hpp

```diff
@@ -0,0 +1,169 @@
+	/// as an ISOBUS shortcut button.
+	///
+	/// This message may be sent by any control function connected to the implement bus on forestry or
+	/// agriculture implements providing to connected systems the current state of the all implement stop operations switch.
+	/// At least one of these switches shall be in each operator location of the connected system.
+	///
+	/// All implements shall start a process to stop all operations when this broadcast message is received from any CF
+	/// with a value of "Stop implement operations" (SPN 5140). Before an implement turns off all implement operations,
+	/// it shall assume a failsafe condition. If an implement is operating in an automation mode,
+	/// it may enter a failsafe condition before requesting the tractor ECU to exit the automation mode,
+	/// e.g. PTO, Auxiliary valve, and/or tractor movement.
+	///
+	/// The working set master for the implement shall then inform the operator that the implement has stopped
+	/// all operations due to the activation of the Stop All Implement Operations switch.
+	/// Implement working set masters shall include, on their home screen, an indication,
+	/// e.g. icon or a function name, if it supports Stop All Implement Operations.
+	/// The Working Set shall monitor the number of transitions for each ISB server upon receiving first
+	/// the message from a given ISB server. A Working Set shall consider an increase in the transitions
+	/// without detecting a corresponding transition of the Stop all implement operations state as an error and react accordingly.
+	class ShortcutButtonInterface : public CANLibProtocol
+	{
+	public:
+		/// @brief Enumerates the states that can be sent in the main ISB message (pgn 64770, 0xFD02)
+		enum class StopAllImplementOperationsState : std::uint8_t
+		{
+			StopImplementOperations = 0, ///< Stop implement operations
+			PermitAllImplementsToOperationOn = 1, ///< Permit all implements to operation ON
+			Error = 2, ///< Error indication
+			NotAvailable = 3 ///< Not available
+		};
+
+		/// @brief Constructor for a ShortcutButtonInterface
+		/// @param[in] internalControlFunction The InternalControlFunction that the interface will use to send messages (not nullptr)
+		/// @param[in] serverEnabled Enables the interface's transmission of the "Stop all implement operations" message.
+		ShortcutButtonInterface(std::shared_ptr<InternalControlFunction> internalControlFunction, bool serverEnabled = false);
+
+		/// @brief Destructor for a ShortcutButtonInterface
+		virtual ~ShortcutButtonInterface();
+
+		/// @brief Adds a callback that will get called when any CF commands a stop to implement operations
+		/// and likewise will be called again when all tracked CFs are sending the "permit implement operations" state
+		/// @param[in] callback The callback to add
+		std::shared_ptr<void> add_stop_all_implement_operations_state_callback(std::function<void(StopAllImplementOperationsState ISBState)> callback);
```

Comment:
It could be better to return the EventDispatcher itself instead here, then we leave the choice of which listener to use (listener with no context, or with safe/unsafe context) up to the user. 

Like here in the HardwareInterface:
https://github.com/ad3154/Isobus-plus-plus/blob/9a790740f3cd26cbd12ffa3493f6fd4be88b80f4/hardware_integration/src/can_hardware_interface.cpp#L223-L236

The VT client still uses the same approach as used here by you, but could be converted to the one in the HardwareInterface as well to make the it more consistent through the project :)

Reply:
Oh, I see what you're saying, yeah I followed along with how the VT did it but this seems more flexible, so I can change it.